PyDynamic/model_estimation/fit_filter.py

```diff
@@ -500,170 +615,140 @@ def invLSFIR_uncMC(H, UH, N, tau, f, Fs, wt=None, verbose=True):
-        Na: int
-            order of IIR denominator polynomial.
-        f: np.ndarray of shape (M,)
-            frequencies corresponding to H
-        Fs: float
-            sampling frequency for digital IIR filter.
-        tau: float
-            initial estimate of time delay for filter stabilization.
+    H : np.ndarray of shape (M,) and dtype complex
+        frequency response values.
+    UH : np.ndarray of shape (2M,2M)
+        uncertainties associated with real and imaginary part of H
+    Nb : int
+        order of IIR numerator polynomial.
+    Na : int
+        order of IIR denominator polynomial.
+    f : np.ndarray of shape (M,)
+        frequencies corresponding to H
+    Fs : float
+        sampling frequency for digital IIR filter.
+    tau : int
+        initial estimate of time delay for filter stabilization.
 
     Returns
     -------
-        b,a: np.ndarray
-            IIR filter coefficients
-        tau: int
-            time delay (in samples)
-        Uba: np.ndarray of shape (Nb+Na+1, Nb+Na+1)
-            uncertainties associated with [a[1:],b]
+    b, a : np.ndarray
+        IIR filter coefficients
+    tau : int
+        time delay (in samples)
+    Uba : np.ndarray of shape (Nb+Na+1, Nb+Na+1)
+        uncertainties associated with [a[1:],b]
 
     References
     ----------
-        * Eichstädt, Elster, Esward and Hessling [Eichst2010]_
+    * Eichstädt, Elster, Esward and Hessling [Eichst2010]_
 
     .. seealso:: :mod:`PyDynamic.uncertainty.propagate_filter.IIRuncFilter`
                  :mod:`PyDynamic.model_estimation.fit_filter.invLSIIR`
     """
-
-    runs = 1000
-
-    print("\nLeast-squares fit of an order %d digital IIR filter to the" % max(Nb, Na))
-    print("reciprocal of a frequency response given by %d values.\n" % len(H))
+    warn(
+        "invLSIIR_unc: The method invLSIIR_unc() will be removed in the next major "
+        "release 3.0.0. Please switch to the current method LSIIR() with the same "
+        "input parameters as in your call of invLSIIR_unc and additionally specify "
+        "inv=True. Just make sure to adapt order or names to the current "
+        "implementation which you will find in the documentation or in the code of "
+        "LSIIR.",
+        PendingDeprecationWarning,
+    )
     print(
-        "Uncertainties of the filter coefficients are evaluated using\n"
-        "the GUM S2 Monte Carlo method with %d runs.\n" % runs
+        f"invLSIIR_unc: Least-squares fit of an order {max(Nb, Na)} digital IIR "
+        f"filter to the reciprocal of a frequency response given by {len(H)} "
+        f"values. Uncertainties of the filter coefficients are evaluated using "
+        "the GUM S2 Monte Carlo method with 1000 runs."
+    )
+    return LSIIR(
+        Hvals=H,
+        Nb=Nb,
+        Na=Na,
+        f=f,
+        Fs=Fs,
+        tau=tau,
+        verbose=False,
+        inv=True,
+        UHvals=UH,
+        mc_runs=1000,
```

Comment:
Alternatively we could as well only raise a DeprecationWarning/Exception and not call LSIIR validly.

Reply:
I would prefer not having (again) a Deprecation Warning for a long time whenever starting PyDynamic. Also in a new release we shouldn't have a deprWarn so early if not really necessary.

Reply:
I chose this pattern for several reasons and had your remark in mind. I am almost certain that you will not get this warning on imprting or using any partof PyDynamic but this specific function. So no annoyance, but no breaking our word as well. Since we state in the old deprecation warning for PyDynamic.deconvolution, that the methods will be found with prefix "inv_" in the new module "model_estimation" form 2.0.0 on. I will check, if you really do not get the deprecation warning until our meeting on friday.

Reply:
We will remove the warning, just leave the correct call in here and open an issue for milestone major release 3.0.0 to consider the removal of this "wrapper".

Reply:
Resolved in 1646a3134f96c4ce54a14149c2be826db79d9acd and with #192 .